packages/data/src/factory.js

```diff
@@ -6,9 +11,25 @@
  * @return {function} marked registry selector.
  */
 export function createRegistrySelector( registrySelector ) {
-	registrySelector.isRegistrySelector = true;
+	const selector = ( ...args ) => registrySelector( selector.registry.select )( ...args );
```

Comment:
It's unclear to me where `selector.registry.select` is coming from as the argument passed to `registrySelector`.  Is it supposed to be `defaultRegistry.select` or I'm _guessing_ it will it be whatever is set on `selector.registry` at the time it's invoked (with a fallback to `defaultRegistry`)?

Reply:
> It's unclear to me where `selector.registry.select` is coming from as the argument passed to `registrySelector`. Is it supposed to be `defaultRegistry.select` or I'm _guessing_ it will it be whatever is set on `selector.registry` at the time it's invoked (with a fallback to `defaultRegistry`)?

It happens here:

https://github.com/WordPress/gutenberg/blob/4edd1d0fcd311101e2db359992384a231fbe88cb/packages/data/src/namespace-store/index.js#L47-L49

This occurs when a store is registered to a registry, so effectively the default registry is **never** used. I added it to simplify / respect the `@type` declaration (tangentially related to #16693), but it could just as well be a noop stub.

Reply:
Makes sense.  I wonder if using a noop stub instead might lead to less head scratching for future looks at this code?

Reply:
> Makes sense. I wonder if using a noop stub instead might lead to less head scratching for future looks at this code?

If we stubbed it, I wouldn't feel comfortable adding `@type {WPDataRegistry}`, since it'd be wrong to claim that the stub is of that type (and I expect potential future type checking could flag it as such).

I'm leaning that the current comment for `selector.registry` assignment alone could be clear enough to explain its purpose.

https://github.com/WordPress/gutenberg/blob/a9e45dc3cdaa7696ac5e686a0ff6fe9eec1f9a5b/packages/data/src/factory.js#L25-L26